docs/docs/mdx/importing-and-using-components.md

```diff
@@ -0,0 +1,64 @@
+---
+title: Importing and Using Components in MDX
+---
+
+You can import your components from other third party libraries that come with components included, like [`theme-ui`](https://theme-ui.com/components). Use cases for external libraries could be charting libraries for adding rich data visualizations, form components for adding email signups, styled portions of content like pullquotes, or buttons for calls to action throughout the content in your pages. You can also import and resuse _your own_ React components and MDX documents.
+
+## Import components for use from another library
+
+Components imported from other libraries can be rendered inline with your markdown content, allowing you to include rich media like charts, interactive buttons, or styled messages. Components are imported at the top of your MDX files—in the same syntax they are imported in JavaScript files—and then added using opening and closing brackets like normal JSX elements.
+
+To include a component from another library (this example uses [the message component from `theme-ui`](https://theme-ui.com/components/message)), you need import it at the top of your MDX file:
+
+```mdx
+---
+title: Importing Components Example
+---
+
+import { Message } from "theme-ui" // highlight-line
+
+You can import your own components.
+
+<Message>MDX gives you JSX in Mardown!</Message> // highlight-line
+```
+
+**Note**: steps for importing custom components or MDX documents from a relative location in your project are also covered in the [Writing Pages in MDX guide](/docs/mdx/writing-pages/).
+
+## Make components available globally as shortcodes
+
+To avoid having to import the same component inside of every MDX document you author, you can add components to an `MDXProvider` to make them globally available in MDX pages. This pattern is sometimes referred to as shortcodes.
+
+```js:title=src/components/layout.js
+import React from "react"
+// highlight-start
+import { MDXProvider } from "@mdx-js/react"
+import { Chart, Pullquote } from "./ui"
+import { Message } from "theme-ui"
+// highlight-end
+
+const shortcodes = { Chart, Pullquote, Message } // highlight-line
+
+export default ({ children }) => (
+  <MDXProvider components={shortcodes}>{children}</MDXProvider> // highlight-line
+)
+```
+
+All MDX components passed into the `components` prop of the `MDXProvider` will be made available to MDX files that are nested under the provider. The `MDXProvider` in this example is in a layout component that would be used to wrap all MDX pages, you can read about this pattern in [the layout section of the `gatsby-plugin-mdx` README](/packages/gatsby-plugin-mdx/#default-layouts).
+
+Now, you can include components in your MDX without importing them:
+
+```mdx
+---
+title: Importing Components Example
+---
+
+You can import your own components.
+
+<Message>MDX gives you JSX in Mardown!</Message> // highlight-line
```

Comment:
```suggestion
<Message>MDX gives you JSX in Mardown!</Message> // highlight-line
```
Wondering if we should just include the chart example to point out that you're using the shortcode?

Reply:
I included both as shortcodes because there were a couple questions on Twitter about where the Chart component was coming from, so I thought maybe including a custom component (the Chart) and a 3rd party component (the Theme UI Message) would show that you can use any component as a shortcode.

Reply:
That seems fair. In that case I think the text about using the short codes needs to come before you use either of them. Does that make sense?